pkg/testing/ess/serverless_provision.go

```diff
@@ -71,114 +62,118 @@ func (prov *ServerlessProvision) SetLogger(l runner.Logger) {
+
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
 
-	gotUp := 0
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
-		case err := <-depErrs:
-			return nil, fmt.Errorf("error waiting for stacks to become available: %w", err)
-		case isUp := <-depUp:
-			if isUp {
-				gotUp++
+			if lastErr == nil {
+				lastErr = ctx.Err()
 			}
-			if gotUp >= len(requests) {
-				return stacks, nil
+			return stack, fmt.Errorf("serverless %s never became ready: %w", stack.Version, lastErr)
+		case <-ticker.C:
+			go func() {
+				statusCtx, statusCancel := context.WithTimeout(ctx, 30*time.Second)
+				defer statusCancel()
+				ready, err := client.DeploymentIsReady(statusCtx)
+				if err != nil {
+					errCh <- err
+				} else if !ready {
+					errCh <- fmt.Errorf("serverless %s never became ready", stack.Version)
+				} else {
+					errCh <- nil
+				}
+			}()
+		case err := <-errCh:
+			if err == nil {
+				stack.Ready = true
+				return stack, nil
 			}
+			lastErr = err
 		}
 	}
-
 }
 
-// Clean shuts down and removes the deployments
-func (prov *ServerlessProvision) Clean(ctx context.Context, stacks []runner.Stack) error {
-	for _, stack := range stacks {
-		prov.stacksMut.RLock()
-		// because of the way the provisioner initializes,
-		// we can't guarantee that we have a valid client/stack setup, as we might have just re-initialized from a file.
-		// If that's the case, create a new client
-		stackRef, ok := prov.stacks[stack.ID]
-		prov.stacksMut.RUnlock()
-		// we can't reference the client, it won't be created when we just run mage:clean
-		// instead, grab the project ID from `stacks`, create a new client
-		if ok {
-			err := stackRef.client.DeleteDeployment()
-			if err != nil {
-				prov.log.Logf("error removing deployment: %w", err)
-			}
-		} else {
-			// create a new client
-			client := NewServerlessClient(prov.cfg.Region, "observability", prov.cfg.APIKey, prov.log)
-			dep_id, ok := stack.Internal["deployment_id"]
-			if !ok {
-				return fmt.Errorf("could not find deployment_id for serverless")
-			}
-			dep_id_str, ok := dep_id.(string)
-			if !ok {
-				return fmt.Errorf("deployment_id is not a string: %v", dep_id)
-			}
-			client.proj.ID = dep_id_str
-
-			dep_type, ok := stack.Internal["deployment_type"]
-			if !ok {
-				return fmt.Errorf("could not find deployment_type in stack for serverless")
-			}
-			dep_type_str, ok := dep_type.(string)
-			if !ok {
-				return fmt.Errorf("deployment_type is not a string: %v", dep_id_str)
-			}
-			client.proj.Type = dep_type_str
-			err := client.DeleteDeployment()
-			if err != nil {
-				return fmt.Errorf("error removing deployment after re-creating client: %w", err)
-			}
+// Delete deletes a stack.
+func (prov *ServerlessProvision) Delete(ctx context.Context, stack runner.Stack) error {
+	deploymentID, deploymentType, err := prov.getDeploymentInfo(stack)
+	if err != nil {
+		return fmt.Errorf("failed to get deployment info from the stack: %w", err)
+	}
```

Comment:
[Suggestion]
```suggestion
	projectID, projectType, err := prov.getprojectInfo(stack)
	if err != nil {
		return fmt.Errorf("failed to get project info from the stack: %w", err)
	}
```

Reply:
+1

Reply:
Why would I change to project? We call the other one deployments, its stored as `deployment_id` and `deployment_type` in the internal context. So confused on this suggestion.

Reply:
Anderson is right, the "official" name is a Serverless project, separate from a Stateful deployment. I don't have a good idea for a term that accurately describes both besides something like "instance".

<img width="1153" alt="Screenshot 2023-11-09 at 3 23 58 PM" src="https://github.com/elastic/elastic-agent/assets/3466215/46341a2f-c9bb-4803-9dab-e41d1895d612">

Reply:
Same as my other comment. I also don't see why "deployment" cannot be used...

Reply:
> Why would I change to project? We call the other one deployments, its stored as deployment_id and deployment_type in the internal context. So confused on this suggestion.

Project is technically the correct term, so the confusion from the above comment is removed. This rename is a completely valid suggestion.

The common term for both could be deployment in contexts that are not serverless specific and probably not cause too much confusion. I don't think we need to block the entire PR on this suggestion, but project is definitely the right word here.

Reply:
As I said in my 1st comment, it's a suggestion, I prefer to have it consistent with how we talk and what we see in the UI. But it isn't a blocker either